### PR TITLE
Update relative outputs regex with shard. [BA-6334]

### PR DESCRIPTION
**Pull request by rhpvorderman**
Shard's are also in the path.

Also the zero `0` should be included in the attempt and shard regexes. In case the number of attempts is higher than 9.
I found this error while running the dev version of cromwell.

### CODE REVIEW

**Comment by rhpvorderman** [comment]
@cjllanwarne, I am sorry to bother you with this. #5437 did not fix the output directory structure completely as shards were overlooked. Any other things that might have been forgotten?

**Comment by rhpvorderman** [comment]
This has been tested in production now for quite some time and it seems that the update catches all folders. (At least on the sfsBackend.)

**Comment by rhpvorderman** [comment]
It is a bit unfortunate cromwell 50 was released before this was merged. It will break all testing everywhere since there is no way of running cromwell and knowing beforehand where the outputs will end up.
In biowdl all testing is already pinned to cromwell-48 to ensure continued operation, we were hoping we could unpin this with 50, but it seems we have to wait a little longer.

File: CHANGELOG.md [hunk 1]
```diff
@@ -11,6 +11,9 @@ Cromwell's metadata archival configuration has changed in a backwards incompatib
 please see
 [the updated documentation](https://cromwell.readthedocs.io/en/stable/Configuring#hybrid-metadata-storage-classic-carbonite) for details.
 
+### Bugfixes
+
++ Fixed a bug where the `use_relative_output_paths` option would not preserve intermediate folders.
```
Comment (aednichols) [inline CHANGELOG.md:16]: Unfortunately this comment must now be relocated to the 51 release notes.

We apologize you didn't make the release, was not on our radar.

Comment (rhpvorderman) [reply CHANGELOG.md]: I have updated the release note.

No need to apoligize. The cromwell team is doing so much already. It's a lot of work to maintain a worklow engine. I don't take that for granted :slightly_smiling_face: . 
Would it help if I attach the "on deck for review" label myself next time I make a small bug or documentation fix? These should be pretty low effort to review and merge.

Comment (aednichols) [reply CHANGELOG.md]: We have a weekly meeting where we select the highest priority PRs and assign them "on deck for review", so self-assignment would not be compatible with that process 😄

Comment (rhpvorderman) [reply CHANGELOG.md]: I will attach the HIGH PRIORITY label then :wink:. But only for small bugfixes like this. I promise I won't be annoying :innocent: .

**Comment by rhpvorderman** [comment]
Thanks!

**Pull request merged**